### PR TITLE
replace bbhx.utils.transform and fix t_offset

### PR DESCRIPTION
**Pull request by WuShichao**
@ahnitz @spxiwh @ConWea This PR replaces the `bbhx.utils.transform` by using the latest development from https://github.com/gwastro/pycbc/pull/4289 and also corrects the offset time when doing relbin PE (the position of LISA in BBHx is not at the right position, not behind the Earth by 20 degrees, we need to correct its position for our multiband PE).

### CODE REVIEW

**Comment by ahnitz** [comment]
@WuShichao This has a conflict with the base plugin? can you rebase and verify that everything still works as expected. (also check why a conflict arose so you understand what the changes were).

File: BBHX_PhenomD.py [hunk 1]
```diff
@@ -68,16 +68,44 @@ def bbhx_fd(ifos=None, run_phenomd=True,
...
-    f_ref = 0.0 # This is now NOT standard LAL convention!
     inc = np.float64(params['inclination'])
-    lam = np.float64(params['eclipticlongitude'])
-    beta = np.float64(params['eclipticlatitude'])
-    psi = np.float64(params['polarization'])
-    t_ref = np.float64(params['tc'])
+    dist = np.float64(pnutils.megaparsecs_to_meters(params['distance']))
+    f_ref = np.float64(params['f_ref'])
+    phi_ref = np.float64(params['coa_phase']) # phase at f_ref
+    t_offset = np.float64(params['t_offset']) # in seconds
```
Comment (ConWea) [inline BBHX_PhenomD.py:75]: Is there a default value set for t_offset? If we require this as a default parameter should we add a warning when not provided?

**Review by ConWea — COMMENTED** [review]
Looks good to me! Only one comment on t_offset.

File: BBHX_PhenomD.py (hunk 1 above)
Comment (WuShichao) [reply BBHX_PhenomD.py:75]: Good spot. We do have a default value for `t_offset`. It's in the multiband frame transform PR https://github.com/gwastro/pycbc/pull/4289/files#diff-7a1da599956f99d94f2c70707533b585de4d61399182d8b58954fe73ed1e15dcR46 which puts LISA to the 20 degrees behind our Earth. I prefer not to hard code an offset in the plugin; just raise a warning/suggestion to use this default value if it's not been set.

Comment (WuShichao) [reply BBHX_PhenomD.py]: I have added a warning, please check.

Comment (ConWea) [reply BBHX_PhenomD.py]: Looks good!

**Review by ConWea — APPROVED** [review]
Looks good to me 👍

**Pull request merged**